set-timeskip-duration.lua

```diff
@@ -0,0 +1,213 @@
+    if seconds > 0 then
+      print("  " .. tostring(seconds) .. " second" .. (seconds == 1 and "" or "s"))
+    end
+  end
+end
+
+function getTargetDate(ticks)
+  local targetYear = tonumber(df.global.cur_year)
+  local targetTick = tonumber(df.global.cur_year_tick)
+  if targetTick + ticks >= 403200 then
+    ticks = ticks - (403200 - targetTick)
+    targetTick = 0
+    targetYear = targetYear + 1
+  end
+  targetYear = targetYear + math.floor(ticks/403200)
+  targetTick = targetTick + (ticks%403200)
+  return targetYear, targetTick
+end
+
+function setTargetDate(scr, ticks) -- df.viewscreen_update_regionst
+  local targetYear, targetTick = getTargetDate(ticks)
+  scr.year = targetYear
+  scr.year_tick = targetTick
+end
+
+local validArgs = utils.invert({
+  'ticks',
+  'years',
+  'months',
+  'days',
+  'hours',
+  'clear',
+  'help'
+})
+local args = utils.processArgs({...}, validArgs)
+
+if args.help then
+  print(usage)
+  return
+end
+
+if args.clear then
+  dfhack.onStateChange.TimeskipMonitor = nil
+  print("Timeskip duration reset to default.")
+  return
+end
+
+local ticks = 0
+
+if args.ticks then
+  if tonumber(args.ticks) and tonumber(args.ticks) >= 0 then
+    ticks = ticks + tonumber(args.ticks)
+  else
+    qerror("Invalid -ticks duration: " .. tostring(args.ticks))
+  end
+end
+
+if args.years then
+  if tonumber(args.years) and tonumber(args.years) >= 0 then
+    ticks = ticks + tonumber(args.years)*403200
+  else
+    qerror("Invalid -years duration: " .. tostring(args.years))
+  end
+end
+
+if args.months then
+  if tonumber(args.months) and tonumber(args.months) >= 0 then
+    ticks = ticks + tonumber(args.months)*33600
+  else
+    qerror("Invalid -months duration: " .. tostring(args.months))
+  end
+end
+
+if args.days then
+  if tonumber(args.days) and tonumber(args.days) >= 0 then
+    ticks = ticks + tonumber(args.days)*1200
+  else
+    qerror("Invalid -days duration: " .. tostring(args.days))
+  end
+end
+
+if args.hours then
+  if tonumber(args.hours) and tonumber(args.hours) >= 0 then
+    ticks = ticks + tonumber(args.hours)*50
+  else
+    qerror("Invalid -hours duration: " .. tostring(args.hours))
+  end
+end
+
+if ticks == 0 and not (args.ticks and tonumber(args.ticks) == 0) then -- permit users to remove the timeskip altogether by setting it to 0 ticks
+  qerror("Duration not specified! Enter \"set-timeskip-duration -help\" for more information.")
+end
+
+ticks = math.floor(ticks) -- get rid of decimals the user may have inputted
+
+printTimeskipCalendarDuration(ticks)
+
+local scr = dfhack.gui.getCurViewscreen()
+if scr._type == df.viewscreen_update_regionst then
+  setTargetDate(scr, ticks)
+else
+  dfhack.onStateChange.TimeskipMonitor = function(event)
+    if event == SC_VIEWSCREEN_CHANGED then
+      local scr = dfhack.gui.getCurViewscreen()
+      if scr._type == df.viewscreen_update_regionst then
+        setTargetDate(scr, ticks)
+        dfhack.onStateChange.TimeskipMonitor = nil
```

Comment:
I noticed that this logic is in place to make the script work at most once per invocation - in my case, it caught me off guard because I had loaded one world, then realized it was the wrong world and decided to load a different world instead, and wound up advancing the second world 14 days instead of the time I expected.

A couple suggestions I thought of - what do you think is best here?

1. Document this behavior (I don't see it documented above, unless I missed something)
2. Add a print statement here, similar to a couple lines below, to indicate that the handler was removed
3. Add a `-persist` argument to make settings persist until explicitly cleared with `-clear`
4. Make settings persist by default until explicitly cleared with `-clear`

Some could be combined (e.g. 1 & 2, 1 & 3), and some are mutually exclusive (3 & 4). We probably can't satisfy everyone, but I just think the current behavior could be a bit more clear.

Reply:
Option (4) sounds best to me, especially since we've already got a `-clear` option in place.